biz.aQute.bndlib/src/aQute/bnd/service/reporter/ReportImportDeserializerPlugin.java

```diff
@@ -0,0 +1,31 @@
+package aQute.bnd.service.reporter;
+
+import java.io.InputStream;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * This plugin deserialize a stream of a specific format into a DTO
+ * representation.
+ */
+@ProviderType
+public interface ReportImportDeserializerPlugin {
+
+	/**
+	 * Get the set of file extension names corresponding to the format that this
+	 * plugin can deserialize.
+	 * 
+	 * @return one or multiple extensions name, never {@code null}
+	 */
+	public String[] getHandledExtensions();
+
+	/**
+	 * Deserialize the input stream into a DTO representation.
+	 * 
+	 * @param input the stream to deserialize, must not be {@code null}
+	 * @return a DTO representation of the input stream content or {@code null}
+	 *         if the stream is empty
+	 * @throws Exception if any errors occur during the deserialization process
+	 */
+	public Object deserialyze(InputStream input) throws Exception;
```

Comment:
I would have thought the method would take the DTO type to create. Something like.

```java
public <D> D deserialize(Class<D> type, InputStream input);
```

Otherwise, what is the type of the returned object? Or are you planning on encoding that into the serialized form like Java serialization?

Reply:
Same here, this plugin is made to import a file and to return a DTO which is collected for the report, then the report will be serialized (with a serializer plugin).